br/pkg/registry/registration.go

```diff
@@ -802,3 +836,57 @@ func (r *Registry) transitionStaleTaskToPaused(ctx context.Context, taskID uint6
 
 	return transitioned, nil
 }
+
+// OperationAfterWaitIDs do the specified operations until the resetting tasks is removed
+func (r *Registry) OperationAfterWaitIDs(ctx context.Context, fn func() error) error {
+	for ids := range slices.Chunk(r.waitIDs, 10) {
+		idStrs := make([]string, 0, len(ids))
+		for _, id := range ids {
+			idStrs = append(idStrs, fmt.Sprintf("%d", id))
+		}
+		idsStr := strings.Join(idStrs, ",")
+		lookupSQL := fmt.Sprintf(selectLeftTasksSQLTemplate, RestoreRegistryDBName, RestoreRegistryTableName, idsStr)
+		for {
+			rows, _, err := r.se.GetSessionCtx().GetRestrictedSQLExecutor().ExecRestrictedSQL(
+				kv.WithInternalSourceType(ctx, kv.InternalTxnBR),
+				nil,
+				lookupSQL,
+				selectLeftTasksSQLTemplate,
+				RestoreRegistryDBName, RestoreRegistryTableName, idsStr,
+			)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			if len(rows) == 0 {
+				break
+			}
+			leftId := rows[0].GetUint64(0)
+			log.Info("wait for the task finishing resetting", zap.Uint64("task id", leftId))
+			time.Sleep(5 * time.Second)
```

Comment:
let's perhaps give this waiting loop a max timeout, the resetting shouldn't be too long but just in case we don't want to stuck here waiting forever, what do you think?